weldx/asdf/util.py

```diff
@@ -288,3 +288,69 @@ def view_tree(file: types_path_and_file_like, path: Tuple = None, **kwargs):
 def asdf_json_repr(file: Union[str, Path, BytesIO], path: Tuple = None, **kwargs):
     """See `view_tree` function."""
     return view_tree(file, path, **kwargs)
+
+
+def dataclass_serialization_class(
+    class_type: Type,
+    class_name: str,
+    version: str,
+    to_tree_mod: Callable = None,
+    from_tree_mod: Callable = None,
+) -> Type:
+    """Generate a asdf serialization class for a python dataclass.
+
+    Parameters
+    ----------
+    class_type :
+        The type of the dataclass
+    class_name :
+        The value that should ba stored as the classes name property
+    version :
+        The version number
+    to_tree_mod :
+        A method that applies additional modifications to the tree during the `to_tree`
+        function call
+    from_tree_mod :
+        A method that applies additional modifications to the tree during the
+        `from_tree` function call
+
+    Returns
+    -------
+    Type :
+        A new asdf serialization class.
+
+    """
+    from copy import deepcopy
+
+    import numpy as np
+
+    from weldx.asdf.types import WeldxType
```

Comment:
any reason these have to be imported here instead of at the top of the file?

Reply:
Not really, think we can move it to the top